### ISSUE TITLE
Syntax highlighting does not work when using clr: prefix

### ISSUE THREAD

**Issue by rohitab** [colorer/FarColorer#73]
### Far Manager version
`3.0.6157.0 x64`

### FarColorer version
`1.4.16`

### Steps to reproduce
Open any file using `clr:` prefix. For example `clr:%FARHOME%\FarEng.lng`

### Expected behavior
File is displayed with syntax highlighting (colors), similar to what you see when you open it in the Editor.

### Actual behavior
File is displayed _without_ syntax highlighting. All text is white on a blue background. Note that syntax highlighting works in the Far Editor. For example, if you open the same file using `edit:%FARHOME%\FarEng.lng`, you can see syntax highlighting.

### Working Version
This feature was working correctly in Far version [`v3.0.5902.2455`](https://github.com/FarGroup/FarManager/releases/tag/ci/v3.0.5902.2455) but stopped working in [`v3.0.5902.2463`](https://github.com/FarGroup/FarManager/releases/tag/ci/v3.0.5902.2463). In FarColorer, it was working correctly in `1.4.10` and stopped working in `1.4.11`.

### Additional Information
The issue appears to be caused due to the changes made in https://github.com/colorer/Colorer-library/commit/76f86119fbd8cba8381543997f72401a11aa83bc. The call to [`setFileType`](https://github.com/colorer/Colorer-library/blob/059c78296a7934ee6b3748ce0309dcbf56ba1a1d/src/colorer/editor/BaseEditor.cpp#L173) was removed from [`BaseEditor::chooseFileType`](https://github.com/colorer/Colorer-library/blob/059c78296a7934ee6b3748ce0309dcbf56ba1a1d/src/colorer/editor/BaseEditor.cpp#L157-L175) method.

```diff
commit 76f86119fbd8cba8381543997f72401a11aa83bc
Author: Aleksey Dobrunov <ctapmex@gmail.com>
Date:   Tue Oct 12 23:11:34 2021 +0500

    don`t set file type when choose him.

diff --git a/src/colorer/editor/BaseEditor.cpp b/src/colorer/editor/BaseEditor.cpp
index a3df039..a8a61ed 100644
--- a/src/colorer/editor/BaseEditor.cpp
+++ b/src/colorer/editor/BaseEditor.cpp
@@ -170,7 +170,6 @@ FileType* BaseEditor::chooseFileType(const UnicodeString* fileName)

     currentFileType = chooseFileTypeCh(fileName, chooseStr, chooseLen);
   }
-  setFileType(currentFileType);
   return currentFileType;
 }
```

### Far Editor
Syntax highlighting works correctly in the Far Editor. The Editor calls `setFileType` after calling `BaseEditor::chooseFileType` as seen in the following code.

https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditor.cpp#L59-L63

The `setFileType` function calls `BaseEditor::setFileType` as seen here.

https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditor.cpp#L65-L67

### Built-in Viewer
The FarColorer built-in viewer does _not_ call `BaseEditor::setFileType` after `BaseEditor::chooseFileType`, as seen in the following code.

https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditorSet.cpp#L203-L204

### Possible Fix
Based on the code called by the Editor, I added a call to `setFileType` in [`FarEditorSet::viewFile`](https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditorSet.cpp#L182-L227) immediately after calling [`chooseFileType`](https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditorSet.cpp#L203). After making this change, the file is syntax highlighted when using the `clr:` prefix. See patch for exact changes.

### Patch
```patch
diff --git a/src/FarEditorSet.cpp b/src/FarEditorSet.cpp
index 23fe195..0653ddf 100644
--- a/src/FarEditorSet.cpp
+++ b/src/FarEditorSet.cpp
@@ -200,7 +200,8 @@ void FarEditorSet::viewFile(const UnicodeString& path)
       regionMap = parserFactory->createStyledMapper(&DConsole, nullptr);
     }
     baseEditor.setRegionMapper(regionMap.get());
-    baseEditor.chooseFileType(&path);
+    FileType* ftype = baseEditor.chooseFileType(&path);
+    baseEditor.setFileType(ftype);
 
     FileType* def = parserFactory->getHrcLibrary().getFileType(UnicodeString(name_DefaultScheme));
     int maxBlockSize = def->getParamValueInt(UnicodeString(param_MaxBlockSize), 300);

```

**Comment by rohitab** [comment]
I created a binary patch to fix this issue. I'm posting it here in case anyone is interested. The patch makes the above code changes to the plugin binary, without requiring you to build it from source.

This patch only is for FarColorer [`v1.4.16 x64`](https://github.com/colorer/FarColorer/releases/tag/v1.4.16)

### Binary Patch
Extract `FarColorer-v1.4.16.patch` from the archive [**`FarColorer-v1.4.16.patch.zip`**](https://github.com/colorer/FarColorer/files/11641814/FarColorer-v1.4.16.patch.zip) to the `%TEMP%` directory, then run the following command in `%FARHOME%\Plugins\FarColorer` directory.

```batch
git apply %TEMP%\FarColorer-v1.4.16.patch
```

**Note**: `FarColorer-v1.4.16.patch` includes the patch for #74 as well. If you want to apply the patches separately, see the following section.

### Manual Patch
Make the following changes to `bin/colorer.dll` using any Hex editor, like [Hexitor](https://sourceforge.net/projects/farplugs/files/Hexitor/).

| File Offset | Original Bytes | Patched Bytes |
|--------|--------|--------|
| `16dd3` | `0F 84 E1 02 00 00` | `90 90 90 90 90 90` |
| `16efd` | `E8 3E 73 01` | `E9 B8 01 00` |
| `170ba` | `48 8D 15 97 DC 2D 00 48 8D 8C 24 30 02 00` | `E8 81 71 01 00 48 89 C2 48 8D 8C 24 60 01` | 
| `170c9` | `E8 12 21 01 00 48 8D 15 83 F1 31 00` | `00 E8 61 82 01 00 E9 2E FE FF FF 90` |

**Comment by ctapmex** [comment]
fixed in FarColorer v1.4.17
